autoinit/components/weighted_sum.py

```diff
@@ -23,27 +23,22 @@ class WeightedSum(tfkeras.layers.Layer):
     a coefficient before summing them together.
     """
 
-    def __init__(self, num_inputs: int = 2, **kwargs):
+    def __init__(self, coefficients=None, **kwargs):
         """
         The constructor initializes the arguments and the
         base class.
         :param num_inputs: Number of inputs to this Layer
         """
         super().__init__(**kwargs)
-        assert num_inputs >= 2
-        self.num_inputs = num_inputs
+        self.coefficients = coefficients
```

Comment:
The coefficients need to be preserved as state, not used as weights.  With state, the same coefficients can be maintained when the layer is serialized/deserialized.  With weights, the weights were being forgotten and were re-initialized as ones.